src/pykeen/nn/modules.py

```diff
@@ -513,19 +513,66 @@ class TransEInteraction(NormBasedInteraction[FloatTensor, FloatTensor, FloatTens
 
 
 @parse_docdata
-class TransFInteraction(FunctionalInteraction[FloatTensor, FloatTensor, FloatTensor]):
-    """A stateless module for the TransF interaction function.
+class TransFInteraction(Interaction[FloatTensor, FloatTensor, FloatTensor]):
+    r"""The state-less norm-based TransF interaction function.
 
-    .. seealso:: :func:`pykeen.nn.functional.transf_interaction`
+    It is given by
+
+    .. math ::
+        f(\mathbf{h}, \mathbf{r}, \mathbf{t}) =
+            (\mathbf{h} + \mathbf{r})^T \mathbf{t} + \mathbf{h}^T (\mathbf{r} - \mathbf{t})
+
+    for head entity, relation, and tail entity representations $\mathbf{h}, \mathbf{r}, \mathbf{t} \in \mathbb{R}^d$.
+    The interaction function can be simplified as
+
+    .. math ::
+        f(\mathbf{h}, \mathbf{r}, \mathbf{t}) &=&
+            (\mathbf{h} + \mathbf{r})^T \mathbf{t} + \mathbf{h}^T (\mathbf{t} - \mathbf{r}) \\
+            &=&
+            \langle \mathbf{h}, \mathbf{t}\rangle
+            + \langle \mathbf{r}, \mathbf{t}\rangle
+            + \langle \mathbf{h}, \mathbf{t}\rangle
+            - \langle \mathbf{h}, \mathbf{r}\rangle \\
+            &=&
+            2 \cdot \langle \mathbf{h}, \mathbf{t}\rangle
+            + \langle \mathbf{r}, \mathbf{t}\rangle
+            - \langle \mathbf{h}, \mathbf{r}\rangle
+
+    .. note ::
+        This is the *balanced* variant from the paper.
+
+    .. todo ::
```

Comment:
is this a new interaction function or a variant inside transf?

Reply:
It is proposed in the same [paper](https://aaai.org/papers/64-12887-knowledge-graph-embedding-by-flexible-translation/), but it is not very precise about that

![Screenshot 2024-10-19 094838](https://github.com/user-attachments/assets/139d0998-afd6-4295-8a3a-a33797f8f4ac)

Reply:
seems like a different one entirely to me. maybe leave as a todo for now